### PR TITLE
Adding ImageSet

### PR DESCRIPTION
**Pull request by joshwalawender**
Hi @eteq, I fooled around with S^4 a bit.  I've added ImageSet object to hold exposure information and added a method to create an ObservingBlock from a list of ImageSets.  This is far from complete, but I ran it through a simple test adapted from your notebook and it seemed to work.

I also had to kludge a fix when running the scheduler failed after a TransitionBlock was added to the schedule.  Not sure if this is the best way to do it, but it worked.

### CODE REVIEW

**Comment by joshwalawender** [comment]
I've now added a new SummingScheduler object which uses the sum of constraint results instead of the product, but this still needs to have the concept of a veto (constraint result = 0) handled properly.  These last couple of commits also add weights for each constraint (to balance the power of one constraint over another) and overall priorities for each ObservingBlock.

File: astroplan/scheduling.py [hunk 1]
```diff
@@ -293,11 +448,15 @@ def __call__(self, oldblock, newblock, start_time, observer):
...
-            components['slew_time'] = sep / self.slew_rate
+            if type(oldblock) == ObservingBlock:
+                aaz = _get_altaz(Time([start_time]), observer, [oldblock.target, newblock.target])['altaz']
+                # TODO: make this [0] unnecessary by fixing _get_altaz to behave well in scalar-time case
+                sep = aaz[0].separation(aaz[1])[0]
+                components['slew_time'] = sep / self.slew_rate
+            elif type(oldblock) == TransitionBlock:
+                # if the previous block was a TransitionBlock, then assume slew
+                # hapens during that block
+                components['slew_time'] = 0.
```
Comment (eteq) [inline astroplan/scheduling.py:459]: Should probably figure out the slew time and then use that to decide if the transition block was long enough, right?

File: astroplan/scheduling.py [hunk 2]
```diff
@@ -34,13 +80,11 @@ def __repr__(self):
...
-    def from_exposures(cls, target, timeperexp, nexp, readouttime=0*u.second,
-                            configuration={}):
-        duration = nexp*(timeperexp + readouttime)
-        ob = cls(target, duration, configuration)
-        ob.timeperexp = timeperexp
-        ob.nexp = nexp
-        ob.readouttime = readouttime
+    def from_imagesets(cls, target, imageset_list, priority=1.0):
+        duration = 0
+        for IS in imageset_list:
```
Comment (eteq) [inline astroplan/scheduling.py:85]: This is nitpicky, but the traditional pep8 solution for situations where you can't use a keyword is to add an `_`.  I.e. `is`->`is_`, not `is`->`IS`, as the latter makes it look like a class.

(Not a big deal at all, but thought I would mention it)

**Comment by eteq** [comment]
@joshwalawender - more on this in the main, PR, but I'm a little confused by the intent of `ImageSet` as implemented here.  Is the idea that you can _only_ make OBs from ISs? That seems to limit the flexibility somewhat (what if instead what I want to do is set up the AO system or get a spectrum instead of an image or whatever?)  

Also, this seems to remove the machinery that sets the configurationsin the OBs, which I would have thought would make the original scheduler fail.  Or am I misunderstanding?

**Comment by joshwalawender** [comment]
@eteq More on the main PR as you say, but very quickly, it was not the intent to only make OBs from ImageSets, but it was an option to make them from ImageSets.

As for the machinery that sets the configurations in the OBs, yes that probably needs some cleanup.  I didn't check that that all passed through properly.

**Pull request closed without merging**